src/main/java/com/nike/cerberus/service/JwtService.java

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright (c) 2019 Nike, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.nike.cerberus.service;
+
+import com.google.inject.name.Named;
+import com.nike.cerberus.jwt.CerberusJwtClaims;
+import com.nike.cerberus.jwt.CerberusJwtKeySpec;
+import com.nike.cerberus.jwt.CerberusSigningKeyResolver;
+import io.jsonwebtoken.Claims;
+import io.jsonwebtoken.InvalidClaimException;
+import io.jsonwebtoken.Jws;
+import io.jsonwebtoken.JwsHeader;
+import io.jsonwebtoken.JwtException;
+import io.jsonwebtoken.Jwts;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.inject.Inject;
+import javax.inject.Singleton;
+import java.time.OffsetDateTime;
+import java.time.ZoneId;
+import java.util.Date;
+import java.util.Optional;
+
+/**
+ * Service for generating, parsing, and validating JWT tokens.
+ */
+@Singleton
+public class JwtService {
+
+    protected final Logger log = LoggerFactory.getLogger(getClass());
+    private final static String PRINCIPAL_TYPE_CLAIM_NAME = "principalType";
+    private final static String GROUP_CLAIM_NAME = "groups";
+    private final static String IS_ADMIN_CLAIM_NAME = "isAdmin";
+    private final static String REFRESH_COUNT_CLAIM_NAME = "refreshCount";
+
+    private final CerberusSigningKeyResolver signingKeyResolver;
+    private final String environmentName;
+
+    @Inject
+    public JwtService(CerberusSigningKeyResolver signingKeyResolver,
+                      @Named("cms.env.name") String environmentName) {
+        this.signingKeyResolver = signingKeyResolver;
+        this.environmentName = environmentName;
+    }
+
+    /**
+     * Generate JWT token
+     * @param cerberusJwtClaims Cerberus JWT claims
+     * @return JWT token
+     */
+    public String generateJwtToken(CerberusJwtClaims cerberusJwtClaims){
+        CerberusJwtKeySpec cerberusJwtKeySpec = signingKeyResolver.resolveSigningKey();
+        String jwtToken = Jwts.builder()
+                .setHeaderParam(JwsHeader.KEY_ID, cerberusJwtKeySpec.getKid())
+                .setId(cerberusJwtClaims.getId())
+                .setIssuer(environmentName)
+                .setSubject(cerberusJwtClaims.getPrincipal())
+                .claim(PRINCIPAL_TYPE_CLAIM_NAME, cerberusJwtClaims.getPrincipalType())
```

Comment:
I think you might consider making an object for the claims and use Jackson to manage serializing and deserializing it to and from a map rather than manually doing it.